composer/algorithms/label_smoothing/README.md

```diff
@@ -0,0 +1,83 @@
+# 🧈 Label Smoothing
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+`Computer Vision`
+
+Label smoothing modifies the target distribution for a task by interpolating between the target distribution and a another distribution that usually has higher entropy (e.g., the uniform distribution). This typically reduces a model’s confidence in its outputs and serves as a form of regularization.
+
+| ![LabelSmoothing](label_smoothing.png) |
+|:--:
+|*Need a picture.*|
+
+## How to Use
+
+### Functional Interface
+
+TODO(CORY): FIX
+
+```python
+def training_loop(model, train_loader):
+  opt = torch.optim.Adam(model.parameters())
+  loss_fn = F.cross_entropy
+  model.train()
+  
+  for epoch in range(num_epochs):
+      for X, y in train_loader:
+          y_hat = model(X)
+          loss = loss_fn(y_hat, y)
+          loss.backward()
+          opt.step()
+          opt.zero_grad()
+```
+
+### Composer Trainer
+
+TODO(CORY): Fix and provide commentary and/or comments
+
+```python
+from composer.algorithms import XXX
+from composer.trainer import Trainer
+
+trainer = Trainer(model=model,
+                  train_dataloader=train_dataloader,
+                  max_duration='1ep',
+                  algorithms=[
+                  ])
+
+trainer.fit()
+```
+
+### Implementation Details
+
+TODO(CORY): Briefly describe how this is implemented under the hood in Composer.
```

Comment:
```suggestion
`LabelSmoothing` converts targets to a one-hot representation if they are not in that format already and then applies a convex combination with a uniform distribution over the labels using the following formula:
```
smoothed_labels = (targets * (1. - alpha)) + (alpha / n_classes)
```
The functional form of the algorithm (accessible with the function `smooth_labels` in `composer.functional`), simply computes smoothed labels and returns them.

The class form of the algorithm also takes care of setting the targets back to the original (pre-smoothing) targets after the loss is computed so that any calculations done with the targets after computing the loss (ex. training metrics) will use the original targets and not the smoothed targets.
```